pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -231,11 +277,25 @@ func (*Plugin) OnOutboundListener(in *plugin.CallbackListenerInputParams, mutabl
 // Can be used to add additional filters (e.g., mixer filter) or add more stuff to the HTTP connection manager
 // on the inbound path
 func (*Plugin) OnInboundListener(in *plugin.CallbackListenerInputParams, mutable *plugin.CallbackListenerMutableObjects) error {
-	if in.Node.Type != model.Router {
-		// Only care about Router nodes.
+	if in.Node.Type != model.Sidecar {
+		// Only care about sidecar.
 		return nil
 	}
-	// TODO: implementation
+	authnPolicy := model.GetConsolidateAuthenticationPolicy(
+		in.Env.Mesh, in.Env.IstioConfigStore, in.ServiceInstance.Service.Hostname, in.ServiceInstance.Endpoint.ServicePort)
+
+	if mutable.Listener == nil || len(mutable.Listener.FilterChains) != 1 {
+		return fmt.Errorf("expect lister has exactly one filter chain")
```

Comment:
this message will be confusing if listener happens to be nil. i suggest taking out the nil check (listener is guaranteed to be not nil, i will add that comment to the API).